dask/array/core.py

```diff
@@ -2771,11 +2752,14 @@ def retrieve_from_ooc(keys, dsk):
     >>> retrieve_from_ooc(g.keys(), g)  # doctest: +SKIP
     """
 
+    if not dsk_post:
+        dsk_post = {k: k for k in keys}
+
     load_dsk = dict()
-    for each_key in keys:
-        load_key = ('load-%s' % each_key[0],) + each_key[1:]
+    for k in keys:
+        load_key = ('load-%s' % k[0],) + k[1:]
         # Reuse the result and arguments from `store_chunk` in `load_chunk`.
-        load_dsk[load_key] = (load_chunk, each_key,) + dsk[each_key][3:-1]
+        load_dsk[load_key] = (load_chunk, dsk_post[k]) + dsk_pre[k][3:-1]
```

Comment:
nit: I'd write this as a dictionary comprehension, but that's just me. No strong opinions.

```python
load_dsk = {('load-%s' % k[0],) + k[1:]: (load_chunk, dsk_post[k]) + dsk_pre[k][3:-1]
            for k in keys}
```

Reply:
Same story as above. Was trying to keep this easy to review, but sure it can be rewritten.